### PR TITLE
docs: RHIDP-876 Enabling authentication - overview

### PR DESCRIPTION
**Pull request by themr0c**
docs: Enabling authentication - overview

The procedure is intentionally very generic, as each provider will have a detailed procedure.

refs https://issues.redhat.com/browse/RHIDP-876

### CODE REVIEW

File: modules/auth/proc-enabling-authentication.adoc [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[id='assembly-enabling-authentication_{context}']
```
Comment (themr0c) [inline modules/auth/proc-enabling-authentication.adoc:1]:
```suggestion
[id='proc-enabling-authentication_{context}']
```

File: modules/auth/proc-enabling-authentication.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Okta:: `okta`
+OAuth 2 Custom Proxy:: `oauth2Proxy`
+OneLogin:: `onelogin`
+SAML:: `saml`
+
+.Procedure
+
+For each provider, follow the dedicated procedure to:
+
+. Set up the required secrets in your provider.
```
Comment (themr0c) [inline modules/auth/proc-enabling-authentication.adoc:34]:
```suggestion
. Set up in your provider the secrets that {product} requires to communicate with the provider.
```

**Comment by kadel** [comment]
The content is ok. But I'm not sure about adding to Getting started. Would it be more logical to have it in the admin section under "Enabling authentication".

I think that getting started should be only about installing and quick initial setup. Than as the last step of getting started we would be linking to the more detailed chapters in admin section

**Review by kadel — DISMISSED** [review]

**Comment by themr0c** [comment]
@linfraze Can you have a look?

**Comment by themr0c** [comment]
@kadel I converted the document into a procedure and lost your approval. I apologize I have to ask your review again.

@linfraze Can you review? Then we have the required assembly to put the procedures for each provider.

**Review by kadel — APPROVED** [review]

File: modules/auth/assembly-enabling-authentication.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
+[id='enabling-authentication']
```
Comment (linfraze) [inline modules/auth/assembly-enabling-authentication.adoc:1]: @hmanwani-rh does the id need to be `assembly-enabling-authentication` or is it ok as is? We might want to convert this to a title one day.

File: modules/auth/proc-enabling-authentication.adoc [hunk 4]
```diff
@@ -0,0 +1,36 @@
+[id='proc-enabling-authentication_{context}']
+= Enabling authentication in {product}
+
+Authentication within the {product} facilitates user sign-in, identification, and access to external resources.
+It supports multiple authentication providers.
+Usually, set up:
```
[the remark this answers is not in the document]
Comment (linfraze) [reply modules/auth/proc-enabling-authentication.adoc]: The structure is good, but this text still reads oddly in the guide. "Usually, set up:" is incomplete and does not clearly communicate what we are trying to say or what we are listing in the bullets that follow.

File: modules/auth/assembly-enabling-authentication.adoc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+= Enabling authentication in {product}
+
+Authentication within {product} facilitates user sign-in, identification, and access to external resources.
+It supports multiple authentication providers.
+
+Usually, set up:
```
Comment (linfraze) [inline modules/auth/assembly-enabling-authentication.adoc:7]:
```suggestion
Authentication providers are typically used in the following ways:
```

File: modules/auth/assembly-enabling-authentication.adoc [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+Google IAP:: `gcp-iap`
+OIDC:: `oidc`
+Okta:: `okta`
+OAuth 2 Custom Proxy:: `oauth2Proxy`
+OneLogin:: `onelogin`
+SAML:: `saml`
+
+For each provider, follow the dedicated procedure to:
+
+. Set up in your provider the secrets that {product} requires to communicate with the provider.
```
Comment (linfraze) [inline modules/auth/assembly-enabling-authentication.adoc:33]:
```suggestion
. Set up the secrets that {product} requires to communicate with the provider.
```

**Review by linfraze — COMMENTED** [review]
The structure looks good! Just made a couple suggestions to clarify the text. 
Next steps are relocating the GitHub auth provider section (already published) and adding the GitLab auth provider section (pending) to the assembly with `includes`.

File: modules/auth/assembly-enabling-authentication.adoc (hunk 6 above)
Comment (themr0c) [reply modules/auth/assembly-enabling-authentication.adoc:33]: I believe here it is important to distinguish what you are doing on the provider side from what you are doing on the product side. With your suggestion we would loose that.

Or maybe:

```suggestion
. Set up the secrets that {product-short} requires to communicate with the provider, on the provider side, then on {product-short} side.
```

and remove the next line. 

WDYT @linfraze?

File: modules/auth/assembly-enabling-authentication.adoc (hunk 3 above)
Comment (themr0c) [reply modules/auth/assembly-enabling-authentication.adoc:1]: I have looked at some other assemblies in the repository, and they do not have the prefix.

File: modules/auth/assembly-enabling-authentication.adoc (hunk 6 above)
Comment (linfraze) [reply modules/auth/assembly-enabling-authentication.adoc:33]: I do agree. I think it's good practice to start by orienting the user, so my original change was _In the provider settings, set up the secrets..._ but it sounded a little awkward with the lead-in sentence, so I just removed it and figured the details would be shared the associated procedures. Maybe we can say _...in both the provider and {product-short} settings_ ? If that is what we actually mean. Saying _on the [provider/RHDH] side_ is a little informal and vague.

File: modules/auth/assembly-enabling-authentication.adoc [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+. Add the authentication provider secrets in {product}.
```
Comment (themr0c) [inline modules/auth/assembly-enabling-authentication.adoc:34]: @linfraze Keeping clarity, and minimal:
 
```suggestion
. Set up the shared secret that the authentication provider and {product} require to communicate.
```

File: modules/auth/assembly-enabling-authentication.adoc [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+GitLab:: `gitlab`
+Google:: `google`
+Google IAP:: `gcp-iap`
+OIDC:: `oidc`
+Okta:: `okta`
+OAuth 2 Custom Proxy:: `oauth2Proxy`
+OneLogin:: `onelogin`
+SAML:: `saml`
+
+For each provider, follow the dedicated procedure to:
```
Comment (nickboldt) [inline modules/auth/assembly-enabling-authentication.adoc:31]:
```suggestion
For each provider, follow the dedicated procedure to:
```

I assume the user is expected to either google the provider's config details, or CTRL-F in the admin guide for details?

Comment (kadel) [reply modules/auth/assembly-enabling-authentication.adoc:31]: I was expecting that the list above will have links to configuration page (section) for each provider. Is that right @themr0c ?

Comment (themr0c) [reply modules/auth/assembly-enabling-authentication.adoc:31]: Yes, when the procedures for each provider will follow, it will make more sense

Comment (themr0c) [reply modules/auth/assembly-enabling-authentication.adoc:31]: https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/78 will come first, then the others, when we have settled on a structure that satisfies everybody.

Comment (linfraze) [inline modules/auth/assembly-enabling-authentication.adoc:31]:
```suggestion
For each provider that you want to use, follow the dedicated procedure to:
```

Comment (linfraze) [reply modules/auth/assembly-enabling-authentication.adoc:31]: Just to clarify that they don't actually have to follow the procedure for each provider, unless they want to.

**Review by linfraze — APPROVED** [review]
One last suggestion :)

**Review by linfraze — APPROVED** [review]
/lgtm

**Comment by linfraze** [comment]
Does this need a cherrypick to 1.1.x?

**Comment by nickboldt** [comment]
unclear. Referenced JIRAs have no fixversion. @themr0c can you set accurate versions on them so it's clear if this is a fix for 1.1+ or for 1.2+?

**Pull request merged**